consensus/src/runway/backup.rs

```diff
@@ -229,78 +229,70 @@ mod tests {
     const NODE_ID: NodeIndex = NodeIndex(0);
     const N_MEMBERS: NodeCount = NodeCount(4);
 
-    struct Unit {
+    async fn produce_units(rounds: usize, session_id: SessionId) -> Vec<Vec<UncheckedSignedUnit>> {
+        let mut creators = creator_set(N_MEMBERS);
+        let keychains: Vec<_> = (0..N_MEMBERS.0)
+            .map(|id| Keychain::new(N_MEMBERS, NodeIndex(id)))
+            .collect();
+
+        let mut units_per_round = Vec::with_capacity(rounds);
+
+        for round in 0..rounds {
+            let pre_units = create_units(creators.iter(), round as Round);
+
+            let units: Vec<_> = pre_units
+                .iter()
+                .map(|(pre_unit, _)| preunit_to_unit(pre_unit.clone(), session_id))
+                .collect();
+            for creator in creators.iter_mut() {
+                creator.add_units(&units);
+            }
+
+            let mut unchecked_signed_units = Vec::with_capacity(pre_units.len());
+            for (i, (pre_unit, _)) in pre_units.into_iter().enumerate() {
```

Comment:
`pre_units.into_iter().zip(keychains.iter())` should also work and not require explicit indexing.

Reply:
Yep, that looks nicer.